src/mito2/src/worker/handle_open.rs

```diff
@@ -71,18 +95,43 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             self.intermediate_manager.clone(),
         )
         .skip_wal_replay(request.skip_wal_replay)
-        .parse_options(request.options)?
         .cache(Some(self.cache_manager.clone()))
-        .open(&self.config, &self.wal)
-        .await?;
-
-        info!("Region {} is opened", region_id);
+        .parse_options(request.options)
+        {
+            Ok(opener) => opener,
+            Err(err) => {
+                sender.send(Err(err));
+                return;
+            }
+        };
 
-        REGION_COUNT.inc();
+        let regions = self.regions.clone();
+        let wal: crate::wal::Wal<S> = self.wal.clone();
```

Comment:
Do we need this annotation?
```suggestion
        let wal = self.wal.clone();
```